edge-runtime/lib/util.ts

```diff
@@ -29,6 +29,19 @@ export const addBasePath = (path: string, basePath?: string) => {
   return path
 }
 
+// add locale prefix if not present, allowing for locale fallbacks
+export const addLocale = (path: string, locale?: string) => {
+  if (
+    locale &&
+    path.toLowerCase() !== `/${locale.toLowerCase()}` &&
+    !path.toLowerCase().startsWith(`/${locale.toLowerCase()}/`) &&
+    !path.toLowerCase().startsWith(`/${locale.toLowerCase()}-`)
```

Comment:
this line feels like something to handle `nl-NL` case? but then in other parts of the PR we are dropping usage of such prefixes in tests so I'm a bit confused here wether we are attempting to handle those cases or not?

Reply:
Thanks - have removed that part of the condition and added a couple of others that were transplanted from `normalizeLocalizedTarget`